go/api/transaction.go

```diff
@@ -0,0 +1,240 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// #include "../lib.h"
+// void successCallback(void *channelPtr, struct CommandResponse *message);
+// void failureCallback(void *channelPtr, char *errMessage, RequestErrorType errType);
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/valkey-io/valkey-glide/go/api/config"
+	"github.com/valkey-io/valkey-glide/go/api/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+type Transaction struct {
+	*baseClient // Embed baseClient to inherit all methods like Get
+	*GlideClient
+	*GlideClusterClient
+	commands []Cmder
+}
```

Comment:
Embedding the clients this way means that anyone can do
```golang
transaction := NewTransaction(client)
transaction.GlideClient.Get("foo")
transaction.GlideClusterClient.Get("foo")
```
and receive a nil panic at runtime.